pkg/worker/verifier/latest_value_map.go

```diff
@@ -11,22 +11,37 @@ import (
 )
 
 type LatestValueMap struct {
-	topic               string
-	LatestKvByPartition []map[string]string
+	topic string
+	// Latest offset for a given key
+	latestKoByPartition []map[string]int64
+	// Latest value for a given key
+	LatestKvByPartition []map[string][]byte
 }
 
-func (lvm *LatestValueMap) Get(partition int32, key string) (value string, exists bool) {
+func (lvm *LatestValueMap) GetValue(partition int32, key string) (value []byte, exists bool) {
 	if partition < 0 || partition >= int32(len(lvm.LatestKvByPartition)) {
 		log.Panicf("Partition %d out of bounds for latestValueMap of size %d", partition, len(lvm.LatestKvByPartition))
 	}
 	value, exists = lvm.LatestKvByPartition[partition][key]
 	return
 }
 
-func (lvm *LatestValueMap) Insert(partition int32, key string, value string) {
+func (lvm *LatestValueMap) InsertKeyValue(partition int32, key string, value []byte) {
 	lvm.LatestKvByPartition[partition][key] = value
 }
 
+func (lvm *LatestValueMap) GetOffset(partition int32, key string) (offset int64, exists bool) {
+	if partition < 0 || partition >= int32(len(lvm.latestKoByPartition)) {
+		log.Panicf("Partition %d out of bounds for latestValueMap of size %d", partition, len(lvm.latestKoByPartition))
+	}
+	offset, exists = lvm.latestKoByPartition[partition][key]
+	return
+}
```

Comment:
unused?

Reply:
In Go, `func (lvm *LatestValueMap) GetOffset(partition int32, key string) (offset int64, exists bool) `

declares that the return values are `offset int64, exists bool`.

By setting their values, and only using a `return` statement, these values are returned.

I know, it's awful!

Reply:
no, is the function itself used anywhere?

Reply:
Oh, my mistake.

No, not currently.

Reply:
But is there any plan to use it later?

TBH I'm not sure I understand the purpose of `GetValue` as opposed to direct access using square brackets -- won't it crash anyway on out-of-bound for partition, just with a different message? And this one makes even less sense to me as it is not used anywhere.

> I know, it's awful!

I'm okay with it BTW, out parameters are there in PL/SQL and pl/pgsql

Reply:
Removed unused function